wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta4.0/1__add_devfile.sql

```diff
@@ -0,0 +1,250 @@
+CREATE UNIQUE INDEX index_devfile_component_name ON devfile_component (devfile_id, name);
+
+
+-- component command
+CREATE TABLE component_command (
+    devfile_component_id    BIGINT,
+    commands                TEXT NOT NULL
+);
+
+-- constraints & indexes
+ALTER TABLE component_command ADD CONSTRAINT fk_component_command_component_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE INDEX index_commands_component_id ON component_command (devfile_component_id);
+
+-- component arg
+CREATE TABLE component_arg (
+    devfile_component_id    BIGINT,
+    args                    TEXT
+);
+
+--constraints
+ALTER TABLE component_arg ADD CONSTRAINT fk_component_command_arg_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE INDEX index_args_component_id ON component_arg (devfile_component_id);
+
+-- component selector
+CREATE TABLE component_selector (
+    devfile_component_id    BIGINT,
+    selector_key            VARCHAR(255),
+    selector                VARCHAR(255)
+);
+
+-- constraints & indexes
+ALTER TABLE component_selector ADD CONSTRAINT fk_component_selector_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE UNIQUE INDEX index_devfile_component_selector ON component_selector (devfile_component_id, selector_key);
+
+-- devfile endpoint
+CREATE TABLE devfile_endpoint (
+    id                      BIGINT       NOT NULL,
+    name                    VARCHAR(255) NOT NULL,
+    port                    INTEGER      NOT NULL,
+    devfile_component_id    BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_endpoint ADD CONSTRAINT fk_devfile_endpoint_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE UNIQUE INDEX index_devfile_endpoint_component_name ON devfile_endpoint (devfile_component_id, name);
+
+-- devfile endpoint attributes
+CREATE TABLE devfile_endpoint_attributes (
+    devfile_endpoint_id    BIGINT,
+    value                  TEXT,
+    name                   VARCHAR(255)
+);
+-- constraints & indexes
+ALTER TABLE devfile_endpoint_attributes ADD CONSTRAINT fk_devfile_endpoint_attributes_id FOREIGN KEY (devfile_endpoint_id) REFERENCES devfile_endpoint (id);
+CREATE UNIQUE INDEX index_devfile_endpoint_attributes_name ON devfile_endpoint_attributes (devfile_endpoint_id, name);
+
+
+-- devfile component env
+CREATE TABLE devfile_env (
+    id                    BIGINT       NOT NULL,
+    name                  VARCHAR(255) NOT NULL,
+    value                 TEXT,
+    devfile_component_id  BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_env ADD CONSTRAINT fk_devfile_env_id FOREIGN KEY (devfile_component_id) REFERENCES devfile_component (id);
+CREATE UNIQUE INDEX index_devfile_env_component_name ON devfile_env (devfile_component_id, name);
+CREATE INDEX index_component_env_id ON devfile_env (devfile_component_id);
```

Comment:
It is redundant since we have `CREATE UNIQUE INDEX index_devfile_env_component_name ON devfile_env (devfile_component_id, name);`